src/core-sdk-tasks/CalculateTemplateVersions.cs

```diff
@@ -9,26 +11,80 @@ namespace Microsoft.DotNet.Cli.Build
+        //  From that version number, get the BundledTemplateInstallPath, BundledTemplateMajorMinorVersion, and BundledTemplateMajorMinorPatchVersion
         [Required]
-        public string AspNetCorePackageVersionTemplate { get; set; }
+        public ITaskItem [] BundledTemplates { get; set; }
+
+        [Required]
+        public string FullNugetVersion { get; set; }
+
+        [Required]
+        public string ProductMonikerRid { get; set; }
+
+        public string InstallerExtension { get; set; }
+
+        [Required]
+        public int CombinedBuildNumberAndRevision { get; set; }
 
-        [Output]
-        public string BundledTemplateInstallPath { get; set; }
 
+        //  Should be the BundledTemplates with BundledTemplateInstallPath metadata set to the value calculated for that group
         [Output]
-        public string BundledTemplateMajorMinorVersion { get; set; }
+        public ITaskItem [] BundledTemplatesWithInstallPaths { get; set; }
+
+        //  For each group of templates (grouped by TemplateFrameworkVersion), this should be the following
+        //  ItemSpec: NetCore60Templates
+        //  TemplateBaseFilename: dotnet-60templates
+        //  TemplatesMajorMinorVersion: 6.0 (from BundledTemplateMajorMinorVersion from group)
+        //  InstallerUpgradeCode: Guid generated using GenerateGuidFromName, combining TemplateBaseFilename, FullNugetVersion, ProductMonikerRid, and InstallerExtension
+        //  MSIVersion: Result of calling GenerateMsiVersionFromFullVersion logic with CombinedBuildNumberAndRevision and BundledTemplateMajorMinorPatchVersion from template group
 
         [Output]
-        public string BundledTemplateMajorMinorPatchVersion { get; set; }
+        public ITaskItem [] TemplatesComponents { get; set; }
 
         private const int _patchVersionResetOffset = 1;
 
         public override bool Execute()
         {
-            var result = Calculate(AspNetCorePackageVersionTemplate);
-            BundledTemplateInstallPath = result.BundledTemplateInstallPath;
-            BundledTemplateMajorMinorVersion = result.BundledTemplateMajorMinorVersion;
-            BundledTemplateMajorMinorPatchVersion = result.BundledTemplateMajorMinorPatchVersion;
+            var groups = BundledTemplates.GroupBy(bt => bt.GetMetadata("TemplateFrameworkVersion"))
+                .ToDictionary(g => g.Key, g =>
+                {
+                    var itemWithVersion = g.SingleOrDefault(i => i.GetMetadata("UseVersionForTemplateInstallPath").Equals("true", StringComparison.OrdinalIgnoreCase));
+                    if (itemWithVersion == null)
+                    {
+                        throw new InvalidOperationException("Could not find single item with UseVersionForTemplateInstallPath for templates with TemplateFrameworkVersion: " + g.Key);
+                    }
+
+                    return Calculate(itemWithVersion.GetMetadata("PackageVersion"));
```

Comment:
It is using the package version of the template that is marked with `UseVersionForTemplateInstallPath` = true, which is the ASP.NET template.  At least, that's how I was intending to preserve the existing behavior.

Reply:
all templates should use aspnet template version to calculate the folder.  All templates should have UseVersionForTemplateInstallPath. But the input is not the packageversion of the template itself. It should be aspnet template version.

For example:
aspnet template, version 3.1.9  -> folder: 3.1.9
test template version 3.2.3 ->  folder: 3.1.9 since aspnet version is 3.1.9
nunit template version 2.1 (if this is the version we want for 3.x) -> folder: 3.1.9 since aspnet version is 3.1.9